docs: add Electron 31 blog post (#573)
electron/website: blog/electron-31-0.md

If you have any feedback, please share it with us on [Twitter](https://twitter.com/electronjs) or [Mastodon](https://social.lfx.dev/@electronjs), or join our community [Discord](https://discord.com/invite/electronjs)! Bugs and feature requests can be reported in Electron's [issue tracker](https://github.com/electron/electron/issues).

## Notable Changes

### Highlights

- Extended `WebContentsView` to accept pre-existing `webContents` object. [#42319](https://github.com/electron/electron/pull/42319)
- Added support for `NODE_EXTRA_CA_CERTS`. [#41689](https://github.com/electron/electron/pull/41689)
- Updated window.flashFrame(bool) to flash continuously on macOS. [#41391](https://github.com/electron/electron/pull/41391)
- Removed `WebSQL` support [#41868](https://github.com/electron/electron/pull/41868)
- `nativeImage.toDataURL` will preserve PNG colorspace [#41610](https://github.com/electron/electron/pull/41610)

> - Extended `webContents.setWindowOpenHandler` to support manual creation of BrowserWindow. [#41432](https://github.com/electron/electron/pull/41432)

Reviewer: nit: I think getting web Filesystem support should get a shout here potentially. The breaking changes to `nativeImage.toDataURL` and `win.flashFrame` also feel pretty niche enough that I don't know if we need to call them out. Just my 2c, but feel free to merge as-is.

- Extended `WebContentsView` to accept pre-existing `WebContents` instances. [#42319](https://github.com/electron/electron/pull/42319)
- Added a new instance property `navigationHistory` on webContents API with `navigationHistory.getEntryAtIndex` method, enabling applications to retrieve the URL and title of any navigation entry within the browsing history. [#41577](https://github.com/electron/electron/pull/41577) <span style={{fontSize: "small"}}>(Also in [29](https://github.com/electron/electron/pull/41661), [30](https://github.com/electron/electron/pull/41662))</span>

### Breaking Changes

#### Removed: `WebSQL` support

Chromium has removed support for WebSQL upstream, transitioning it to Android only. See
[Chromium's intent to remove discussion](https://groups.google.com/a/chromium.org/g/blink-dev/c/fWYb6evVA-w/m/pziWcvboAgAJ)
for more information.

#### Behavior Changed: `nativeImage.toDataURL` will preseve PNG colorspace

PNG decoder implementation has been changed to preserve colorspace data. The
encoded data returned from this function now matches it.

See [crbug.com/332584706](https://issues.chromium.org/issues/332584706) for more information.

> #### Behavior Changed: `window.flashFrame(bool)` will flash dock icon continuously on macOS

Reviewer: nit: I think the standard way of referencing a BrowserWindow instance in the Electron docs is `win.flashFrame` rather than `window.flashFrame`. This also prevents any confusion with the renderer `window` global.

```suggestion
#### Behavior Changed: `win.flashFrame(bool)` will flash dock icon continuously on macOS
```

If you have any feedback, please share it with us on [Twitter](https://twitter.com/electronjs) or [Mastodon](https://social.lfx.dev/@electronjs), or join our community [Discord](https://discord.com/invite/electronjs)! Bugs and feature requests can be reported in Electron's [issue tracker](https://github.com/electron/electron/issues).

## Notable Changes

### Highlights

- Extended `WebContentsView` to accept pre-existing `webContents` object. [#42319](https://github.com/electron/electron/pull/42319)
- Added support for `NODE_EXTRA_CA_CERTS`. [#41689](https://github.com/electron/electron/pull/41689)
- Updated window.flashFrame(bool) to flash continuously on macOS. [#41391](https://github.com/electron/electron/pull/41391)
- Removed `WebSQL` support [#41868](https://github.com/electron/electron/pull/41868)
- `nativeImage.toDataURL` will preserve PNG colorspace [#41610](https://github.com/electron/electron/pull/41610)

> - Extended `webContents.setWindowOpenHandler` to support manual creation of BrowserWindow. [#41432](https://github.com/el
>   ectron/electron/pull/41432)

Reviewer: ```suggestion
- Extended `webContents.setWindowOpenHandler` to support manual creation of BrowserWindow. [#41432](https://github.com/electron/electron/pull/41432)
```

Don't think this works in Markdown 🤔

- Added options parameter to `Session.clearData` API. [#41355](https://github.com/electron/electron/pull/41355)
- Added support for Bluetooth ports being requested by service class ID in `navigator.serial`. [#41638](https://github.com/electron/electron/pull/41638)
- Added support for `NODE_EXTRA_CA_CERTS`. [#41689](https://github.com/electron/electron/pull/41689)
- Extended `webContents.setWindowOpenHandler` to support manual creation of BrowserWindow. [#41432](https://github.com/electron/electron/pull/41432)
- Implemented support for the File System API. [#41419](https://github.com/electron/electron/pull/41419)
- `clearData` method added to `Session`. [#40983](https://github.com/electron/electron/pull/40983)
- Extended `WebContentsView` to accept pre-existing `webContents` object. [#42319](https://github.com/electron/electron/pull/42319)

### Breaking Changes

#### Removed: `WebSQL` support

Chromium has removed support for WebSQL upstream, transitioning it to Android only. See

> [Chromium's intent to remove discussion](https://groups.google.com/a/chromium.org/g/blink-dev/c/fWYb6evVA-w/m/wGI863zaAAAJ)

Author: Thanks, great catch! Wee note for myself: we'll also need to update this in the `breaking-changes.md` doc

> - Added a new instance property `navigationHistory` on webContents API with `navigationHistory.getEntryAtIndex` method, enabling applications to retrieve the URL and title of any navigation entry within the browsing history. [#41577](https://github.com/electron/electron/pull/41577) <span style="font-size:small;">(Also in [29](https://github.com/electron/electron/pull/41661), [30](https://github.com/electron/electron/pull/41662))</span>

Reviewer: issue(blocking): the website deploy is failing because it expects JSX-style properties, I think.

```suggestion
- Added a new instance property `navigationHistory` on webContents API with `navigationHistory.getEntryAtIndex` method, enabling applications to retrieve the URL and title of any navigation entry within the browsing history. [#41577](https://github.com/electron/electron/pull/41577) <span style={{fontSize: "small"}}>(Also in [29](https://github.com/electron/electron/pull/41661), [30](https://github.com/electron/electron/pull/41662))</span>
```

Reviewer: Confirmed locally that this renders properly, will commit the suggestion.
